### PR TITLE
Add legacy SHC revocation RFC

### PR DESCRIPTION
**Pull request by christianpaquin**
RFC for legacy card revocation, as discussed in #207.

TODO:
 - Depends on PR #205. Update text after merge.
 - Add some of the commentary from discussion #207.
 - Give better guidance on method selection.

### CODE REVIEW

**Review by jmandel — APPROVED** [review]
A couple of notes that needn't prevent merging!

File: rfcs/002-legacy-shc-revocation.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+The following methods can be used by Issuers to create their CRLs. Only one method can be used for each Issuer key (identified by an identifier `kid`), and the method identifier must be listed in the CRL's `method` field. Adding a timestamp to revocation IDs and CRL updates work as specified in the core SHC framework.
+
+Verifiers validating a SHC without a `rid` value for which the issuer key lists a `crlVersion` SHALL download the CRL and use the specified method to calculate and check the revocation status of the SHC.
+
+## FHIR Bundle digest
+
+Method identifier: `hash-fhir`
+
+To use this scheme, an Issuer needs to be able to recreate the FHIR bundle for the revoked SHCs. Given a FHIR bundle, the revocation ID `rid` is computed as the base64url encoding of the first 64 bits of the SHA-256 digest of the base64url encoding of the UTF-8 encoding of the minified FHIR bundle; i.e., `rid = base64url(sha256(base64url(utf8(fhir)))[0..8])`.
```
[the remark this answers is not in the document]
Comment (christianpaquin) [reply rfcs/002-legacy-shc-revocation.md:19]: That's a good point. I like the idea of adding canonicalization (using [RFC8785](https://datatracker.ietf.org/doc/html/rfc8785) rules?)

[the remark this answers is not in the document]
Comment (christianpaquin) [reply rfcs/002-legacy-shc-revocation.md]: Added canonicalization to the proposal.

**Comment by christianpaquin** [comment]
I'm considering removing the `hmac-patient` method from this rfc; doesn't seem to meet the requirements of the issuers we talked to. Anyone planning to use it?

**Comment by christianpaquin** [comment]
FYI @jmandel, since you approved a month ago; I made some updates to link to the revocation artefacts (now merged) in the spec, and I removed the hmac-patient method that no one expressed interest in.

**Review by jmandel — APPROVED** [review]

**Comment by jmandel** [comment]
Thanks! Still looks good, and indeed even better. Now:-) If you have permission to merge you can go ahead. Otherwise I am happy to press the button. Thanks! Still looks good, and indeed even better now :)

I'd say you can go ahead and merge; We can have a pretty low threshold for RFC content at draft status.

**Comment by christianpaquin** [comment]
> I'd say you can go ahead and merge; We can have a pretty low threshold for RFC content at draft status.

I don't have write access to the repo, so I'll let you merge @jmandel.

**Pull request merged**